install.sh

```diff
@@ -167,6 +171,60 @@ legacy_transformations_start() {
     return 0
 }
 
+############################################################
+# Function to dispatch to all legacy transformations
+#   right after the update.
+#
+# Returns: 0 if succesful, 1 otherwise
+############################################################
+legacy_transformations_after_update() {
+    COMMIT_COUNT=$(execute_git "$GITHOOKS_CLONE_DIR" rev-list --count HEAD)
+    CURRENT_COMMIT=$(execute_git "$GITHOOKS_CLONE_DIR" rev-parse HEAD)
+
+    if [ "$COMMIT_COUNT" != "1" ] && [ -z "$INTERNAL_UPDATED_FROM_COMMIT" ]; then
+        # If the clone dir has been updated and
+        # we do not have the update commit yet (meaning we are updating from an old version )
+        # we set it to the last commit where this feature was not yet available
+        INTERNAL_UPDATED_FROM_COMMIT="ab86d2a529f58744a71e79227e434f19b84589e6"
+    fi
+
+    LEGACY_TRANSFORM_MOVE_LOCAL_PATHS="false"
+    LEGACY_TRANSFORM_SPLIT_GLOBAL_SHARED_ENTRIES="false"
+
+    if [ -n "$INTERNAL_UPDATED_FROM_COMMIT" ] &&
+        [ "$CURRENT_COMMIT" != "$INTERNAL_UPDATED_FROM_COMMIT" ]; then
+        # Last commit before the update is set and the current is ahead
+        # Meaning, we can now test if legacy transform checks are needed:
+
+        # Check if we need transforms for:
+        # - .shared files: put local paths into `--local githooks.shared`.
+        # - `--global githooks.shared` is split into
+        #    multiple config values.
+        # which was introduced by PR #125 right after commit ab86d2a5:
+        # ============================================================
+        FEATURE_BEFORE="ab86d2a529f58744a71e79227e434f19b84589e6"
+
+        FEATURE_LE_CURRENT="false"
+        execute_git "$GITHOOKS_CLONE_DIR" merge-base --is-ancestor \
+            "$FEATURE_BEFORE" "$CURRENT_COMMIT" &&
+            [ "$CURRENT_COMMIT" != "$FEATURE_BEFORE" ] &&
+            FEATURE_LE_CURRENT="true" # feature <= current commit
```

Comment:
Is this check really needed? Isn't it implicit that any feature that this install.sh knows about is an ancestor of the current version?

Reply:
Jup, true that! Good point :)